test/pylib/scylla_cluster.py

```diff
@@ -760,6 +768,22 @@ async def add_server(self, replace_cfg: Optional[ReplaceConfig] = None,
         self.logger.info("Cluster %s added %s", self, server)
         return ServerInfo(server.server_id, server.ip_addr)
 
+    async def add_servers(self, servers_num: int = 1,
+                          cmdline: Optional[List[str]] = None,
+                          config: Optional[dict[str, Any]] = None,
+                          property_file: Optional[dict[str, Any]] = None,
+                          start: bool = True,
+                          expected_error: Optional[str] = None) -> [ServerInfo]:
+        """Add multiple servers to the cluster concurrently"""
+        assert servers_num > 0, f"add_servers: cannot add {servers_num} servers"
+
+        # If the cluster is empty, all servers must use the same seed to not start separate clusters.
+        if not self.servers:
+            self.initial_seed = await self._get_new_ip()
```

Comment:
This is leaking IPs if `add_server` below fails, no?

Reply:
Also, instead of having this `is_initial_seed` parameter, wouldn't it be more elegant to lease all IPs up-front here, and pass them to `add_server`? `add_server` would take an IP address as a parameter, forcing the caller to provide IP. Perhaps `add_server` could be turned into a private function `_add_server`, and there would be a simple `add_server` wrapper that obtains the IP and calls `_add_server`. `add_servers` would similarly call `_add_server`. The wrappers would release IP(s) on error.

Reply:
I solved both of these problems by moving `initial_seed` initialization to `add_server`.

But I've just realized that introducing `_get_new_ip` is pointless now since it is and will be used only once. I'll quickly resubmit the PR.